### PR TITLE
Have Kernel Inherit IrContainer

### PR DESCRIPTION
**Pull request by csarofeen**
Move Kernel to inherit IrContainer in a step to move it to inherit from Fusion.
kernel()->irStmts() gets moved to unordered_exprs
Add’s Val::kernel which just dynamically casts if the container in Val is Kernel

Just a warning that the inKernel inFusion and assert variants are moved in a later PR to just be inContainer in IrContainer.
A follow up PR also makes Kernel non-copy, non-move, and removes the default constructor.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp [hunk 1]
```diff
@@ -58,12 +58,20 @@ std::string Statement::toString() const {
...
+
+kir::Kernel* Statement::kernel() const {
+  TORCH_INTERNAL_ASSERT(
+      ir_container_->isA<kir::Kernel>(),
+      "Statement does not belong to a kernel.");
+  return ir_container_->as<kir::Kernel>();
+}
+
+bool Statement::isKirStmt() const {
+  return is_kir_stmt_ || container()->isA<kir::Kernel>();
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp]: Why `is_kir_stmt_` used here? Is it going to be cleaned up?

File: torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp [hunk 2]
```diff
@@ -20,30 +20,42 @@ namespace {
...
+            IrBuilder::create<NamedScalar>(
+                GpuLower::current()->kernel()->as<IrContainer>(),
+                stringifyThread(pt),
+                DataType::Int),
+            simplifying_ir_builder.subExpr(
+                IrBuilder::create<NamedScalar>(
+                    GpuLower::current()->kernel()->as<IrContainer>(),
+                    stringifyThreadSize(pt),
+                    DataType::Int),
+                simplifying_ir_builder.oneVal()))
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp]: The previous code looks better to me as it's more succinct, though I'm fine with the new one as well.

**Review by naoyam — APPROVED** [review]
Added some minor comments. LGTM.

File: torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp]: This gets cleaned up at latest in https://github.com/csarofeen/pytorch/pull/1390

File: torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp (hunk 2 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp]: This does get cleaned up in a later PR, uncertain which one.

**Pull request merged**